### PR TITLE
IEN-674 update ats sync with new data

### PR DESCRIPTION
**Pull request by sunghwan-park-fw**
- Use HMBC_ATS_BASE_URL = `https://static.heabc.bc.ca/ien`
- Change job titles(departments) table primary column type to uuid
- Change health authority table primary column type to uuid
- Change withdraw reason table primary column type to uuid
- Remove milestones deleted from ATS

### CODE REVIEW

**Review by dbayly-freshworks — APPROVED** [review]
The looks pretty good. A couple nits/questions. 
Also, in principle, it may be good to add a down function to these migrations

File: apps/api/src/applicant/ienapplicant.util.service.ts [hunk 1]
```diff
@@ -284,15 +284,14 @@ export class IENApplicantUtilService {
           DESC limit 1
         )
         WHERE ien_applicants.id IN (${idsToUpdate})`;
-      const updatedApplicants = await getManager().query(queryToUpdate);
-      this.logger.log({ updatedApplicants });
+      await getManager().query(queryToUpdate);
```
[the remark this answers is not in the document]
Comment (sunghwan-park-fw) [reply apps/api/src/applicant/ienapplicant.util.service.ts:287]: The log is like this. do you want to restore it?
`[IEN] Info	2022-09-15, 10:36:03 a.m.  - { updatedApplicants: [ [], 1 ] }`

I think we need to standardize the log format by transaction.

File: apps/api/src/applicant/external-api.service.ts [hunk 2]
```diff
@@ -343,6 +345,38 @@ export class ExternalAPIService {
...
   }
 
+  async removeMilestonesNotOnATS(
+    applicants: { applicant_id: string; milestones: { id: string }[] }[],
+  ): Promise<void> {
+    let removedCount = 0;
+    await Promise.all(
+      applicants.map(async a => {
+        const audits: { id: string; status_id: string }[] = await getManager().query(`
+        SELECT "audit"."id" id, "status"."id" status_id
```
Comment (sunghwan-park-fw) [inline apps/api/src/applicant/external-api.service.ts:355]: @dbayly-freshworks @coryatwater-FW 
I wanted to use query builder but it doesn't work. The weird thing is that this query is from the debug output of the query builder.

File: apps/api/src/applicant/ienapplicant.util.service.ts [hunk 3]
```diff
@@ -273,26 +273,27 @@ export class IENApplicantUtilService {
...
+        .orderBy('audit.start_date', 'DESC')
+        .limit(1);
+      const result = await this.ienapplicantRepository
+        .createQueryBuilder('applicant')
+        .update({
+          status: () => '(' + subQuery + ')',
+        })
+        .where({ id: In(mappedApplicantList) })
+        .execute();
+      this.logger.log(`applicants status updated: ${result.affected}`);
```
Comment (sunghwan-park-fw) [inline apps/api/src/applicant/ienapplicant.util.service.ts:289]: @dbayly-freshworks 
I changed the raw query to query builder and restored logging

File: apps/api/src/applicant/ienapplicant.util.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sunghwan-park-fw) [reply apps/api/src/applicant/ienapplicant.util.service.ts]: I tried to use query builder to update the latest milestone but it didn't work as expected.
I just restored the log.

**Pull request merged**